<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Multiscale modeling of HIV</title>
  <meta name="description" content="">
  <meta name="author" content="Simon Frost">
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
  <link rel="stylesheet" href="libraries/frameworks/revealjs/css/reveal.min.css">
  <link rel="stylesheet" href="libraries/frameworks/revealjs/css/theme/simple.css" id="theme">
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" id="theme">
  <!--[if lt IE 9]>
  <script src="lib/js/html5shiv.js"></script>
  <![endif]-->  <link rel="stylesheet" href = "assets/css/msa.css">
<link rel="stylesheet" href = "assets/css/nv.d3.css">
<link rel="stylesheet" href = "assets/css/reveal.css">
<link rel="stylesheet" href = "assets/css/reveal.min.css">
<link rel="stylesheet" href = "assets/css/ribbons.css">
<link rel="stylesheet" href = "assets/css/simple.css">

</head>
<body>
  <div class="reveal">
    <div class="slides">
      <section class='' data-state='blackout' id='slide-1'>
  
  <h2 style="color:#FFFF00">Multiscale modeling of HIV dynamics and evolution</h2>

<p><br>
<br></p>

<h3 style="color:#FFFFFF">Simon Frost, M.A. D.Phil.</h3>

<p><em>Dept. of Veterinary Medicine, and Institute of Public Health</em></p>

<p><em>University of Cambridge</em></p>

<p><img width="10%" style="border:0px;" src="assets/img/cu.svg"></p>

<script src="assets/js/jquery.js"></script>

<p><link rel=stylesheet href="assets/css/nv.d3.css"></link></p>

<script src="assets/js/d3.js"></script>

<script src="assets/js/nv.d3.js"></script>

<script src="assets/js/viruspie.js"></script>

</section>
<section class='' data-state='' id='slide-2'>
  <h2>Why model?</h2>
  <ul>
<li>We often only have observational data

<ul>
<li>Models allow us to ask &#39;what if?, and exploit &#39;natural experiments&#39;</li>
</ul></li>
<li>Typically inexpensive</li>
<li>Models make assumptions explicit

<ul>
<li>Many researchers use models implicitly</li>
</ul></li>
<li>Most powerful when they make <em>testable predictions</em> or are just plain wrong</li>
</ul>

</section>
<section class='' data-state='' id='slide-3'>
  <h2>Why HIV?</h2>
  <div>
<!-- IMAGEMAP START-->
<script>
function Switch(picname,location) {
    document.images[picname].src = location;
}
</script>
<img src="assets/img/aids1.jpg" width="780" height="400" usemap="#aids-imagemap" name="map" border=0 alt="" style="border:0px;"/>
<map name="aids-imagemap">
<area shape="rect" coords="5,13,194,40" href="" onMouseOver="Switch('map','assets/img/aids1.jpg')">
<area shape="rect" coords="198,13,388,40" href="" onMouseOver="Switch('map','assets/img/aids2.jpg')">
<area shape="rect" coords="392,13,582,40" href="" onMouseOver="Switch('map','assets/img/aids3.jpg')">
<area shape="rect" coords="586,13,776,40" href="" onMouseOver="Switch('map','assets/img/aids4.jpg')">
</map>
<!-- IMAGEMAP END-->
</div>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Graphic: CBC Data:UNAIDS</em></p>
</div>

</section>
<section class='' data-state='' id='slide-4'>
  <h2>HIV in the Philippines</h2>
  <p><img src="assets/fig/unnamed-chunk-3-1.png" title="plot of chunk unnamed-chunk-3" alt="plot of chunk unnamed-chunk-3" style=border:0px; /></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Data: Philippines DoH, May 2015</em></p>
</div>

</section>
<section class='' data-state='' id='slide-5'>
  <h2>HIV in the UK</h2>
  <p><img src="assets/img/ukhiv.png" style="border: 0px;" width="70%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>PHE (2014) HIV in the United Kingdom</em></p>
</div>

</section>
<section class='' data-state='' id='slide-6'>
  <h2>HIV drug resistance</h2>
  <ul>
<li>HIV treatment has reduced the number of deaths due to HIV</li>
<li>However, treatment is associated with <strong>acquired drug resistance</strong>

<ul>
<li>Emerges in response to therapy</li>
</ul></li>
<li>At the population level, we also observed <strong>transmitted drug resistance</strong>

<ul>
<li>Individuals are infected with resistant virus</li>
</ul></li>
<li>We can sequence part of the viral genome to tell whether it is resistant or not</li>
</ul>

</section>
<section class='' data-state='' id='slide-7'>
  <h2>HIV is the most sequenced virus</h2>
  <div id="viruspie" width="900px" height="600px" >
    <svg width="800" height="500" style="font-family: 'xkcd';font-size: 20px;"></svg>
</div>

</section>
<section class='' data-state='' id='slide-8'>
  <h2>HIV as a model evolutionary system</h2>
  <ul>
<li>In addition to there being a vast amount of data, HIV can also tell us about evolutionary processes</li>
<li>HIV has a short generation time

<ul>
<li>Population dynamics within the individual can occur on a short timescale</li>
</ul></li>
<li>HIV has a high mutation rate</li>
<li>There is a large number of infected cells within a typical infected person</li>
<li>HIV is <em>measurably evolving</em></li>
</ul>

</section>
<section class='' data-state='' id='slide-9'>
  <h2>The within-host life cycle of HIV</h2>
  <p><img src="assets/img/perelson_1996.png" style="border: 0px;" width="60%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Perelson (1996)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-10'>
  <h2>Multiple scales and HIV</h2>
  <ul>
<li>Within an infected cell</li>
<li>Within pockets of target cells</li>
<li>Within an organ</li>
<li>Between organs</li>
<li>Between pairs of individuals</li>
<li>Within a population of individuals</li>
<li>At the global level</li>
</ul>

</section>
<section class='' data-state='' id='slide-11'>
  <h2>Examples</h2>
  <ul>
<li>Evolution of drug resistance</li>
<li>Evolution of immune escape

<ul>
<li>Humoral responses</li>
<li>Cellular responses</li>
</ul></li>
<li>Molecular epidemiology</li>
</ul>

</section>
<section class='' data-state='' id='slide-12'>
  <h2>Evolution of resistance to AZT</h2>
  <ul>
<li>AZT was the first antiviral agent to be licensed for treatment of HIV</li>
<li>Although CD4+ counts recovered, the virus rebounded</li>
</ul>

<p><br>
<img src="assets/img/aztdynamics.png" style="border: 0px;" width="70%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>McLean and Nowak (1992)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-13'>
  <h2>Resistance higher with higher doses</h2>
  <p><br>
<img src="assets/img/aztresistance.png" style="border: 0px;" width="60%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>McLean and Nowak (1992)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-14'>
  <h2>Resistance and dose</h2>
  <p><img src="assets/fig/unnamed-chunk-4-1.png" title="plot of chunk unnamed-chunk-4" alt="plot of chunk unnamed-chunk-4" style=border:0px; /></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Model from: McLean and Nowak (1992)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-15'>
  <h2>Stochastic versus deterministic</h2>
  <ul>
<li>Treatment response varies between individual</li>
<li>Many biological models are deterministic in nature

<ul>
<li>Model takes the form of differential equations</li>
<li>Variability between individuals reflects biological variation</li>
</ul></li>
<li>Variation could arise simply through chance effects</li>
</ul>

</section>
<section class='' data-state='' id='slide-16'>
  <h2>Evolution of 3TC resistance</h2>
  <ul>
<li>3TC or lamivudine was the second drug to be licensed</li>
<li>High level resistance can be achieved through mutations at a single site in the reverse transcriptase</li>
</ul>

<p><br>
<img src="assets/img/nontams.png" style="border: 0px;" width="40%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p>http://hivdb.stanford.edu/pages/3DStructures/rt.html</p>
</div>

</section>
<section class='' data-state='' id='slide-17'>
  <h2>Evolution of M184I/V</h2>
  <iframe src="m184v.html" style="border:0px;" width="800px" height="600px"></iframe>

</section>
<section class='' data-state='' id='slide-18'>
  <h2>Within-host evolution of M184</h2>
  <p><img src="assets/img/m184v.jpg" style="border: 0px;" width="60%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. J. Virol. 2000</em></p>
</div>

</section>
<section class='' data-state='' id='slide-19'>
  <h2>Rise of M184V</h2>
  <p><img src="assets/img/m184v_line.jpg" style="border: 0px;" width="60%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. J. Virol. 2000</em></p>
</div>

</section>
<section class='' data-state='' id='slide-20'>
  <h2>Mutation-selection balance</h2>
  <ul>
<li>Prior to therapy, resistant mutations exist at <em>mutation-selection balance</em></li>
<li>Population genetics theory tells us that there can be random fluctuations due to genetic drift</li>
<li>The extent of these fluctuations can be captured by the concept of an <em>effective population size</em> or $N_e$</li>
</ul>

</section>
<section class='' data-state='' id='slide-21'>
  <h2>Fluctuations in M184V/I before therapy</h2>
  <p><img src="assets/fig/unnamed-chunk-5-1.png" title="plot of chunk unnamed-chunk-5" alt="plot of chunk unnamed-chunk-5" style="border:0px" /></p>

</section>
<section class='' data-state='' id='slide-22'>
  <h2>Why stochastic?</h2>
  <ul>
<li>There are <em>hundreds of millions</em> of infected cells in a typical HIV-infected person</li>
<li>How can stochasticity be important when the population size is so large?</li>
</ul>

</section>
<section class='' data-state='' id='slide-23'>
  <h2>Within host metapopulation dynamics</h2>
  <p><img src="assets/img/metapop.jpg" style="border: 0px;" width="30%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. PNAS 2001</em></p>
</div>

</section>
<section class='' data-state='' id='slide-24'>
  <h2>Model</h2>
  <ul>
<li>Implications:

<ul>
<li>Repeated founder effects result in low effective population sizes</li>
</ul></li>
</ul>

<p><img src="assets/img/metapop_eqn.gif" style="border: 0px;" width="40%"></p>

<ul>
<li>Predictions:

<ul>
<li>There should be fine-scale spatial structure</li>
<li>Within subpopulations, there should be evidence of founder effects</li>
</ul></li>
</ul>

</section>
<section class='' data-state='' id='slide-25'>
  <h2>Testing for spatial differentiation</h2>
  <ul>
<li>$F_{ST}$: the fraction of genetic variation <em>within</em> subpopulations relative to <em>total</em> variation</li>
<li>Under a metapopulation model:</li>
</ul>

<p><br></p>

<p>$$
F_{ST} = \frac{k+N_s x}{k(1+N_sx)} \approx \frac{1}{k}
$$</p>

</section>
<section class='' data-state='' id='slide-26'>
  <h2>Evidence for micro-scale spatial differentiation</h2>
  <p><br></p>

<div>
<table frame="hsides" rules="groups" id="table-1">
                           <thead id="thead-1">
                              <tr id="tr-1">
                                 <th rowspan="1" colspan="1" id="th-1">Patient</th>
                                 <th rowspan="1" colspan="1" id="th-2">Pulps</th>
                                 <th rowspan="1" colspan="1" id="th-3">Between-pulp
                                    variation</th>
                                 <th rowspan="1" colspan="1" id="th-4">Total variation </th>
                                 <th rowspan="1" colspan="1" id="th-5"><em>F</em><sub>ST</sub>
                                                                   </th>
                              </tr>
                           </thead>
                           <tbody align="center" id="tbody-1" class="table-center">
                              <tr id="tr-2">
                                 <td id="td-1">B</td>
                                 <td id="td-2">3</td>
                                 <td id="td-3">1.453</td>
                                 <td id="td-4">2.446</td>
                                 <td id="td-5">0.594
                                    
                                 </td>
                              </tr>
                              <tr id="tr-3">
                                 <td id="td-6">L</td>
                                 <td id="td-7">4</td>
                                 <td id="td-8">0.836</td>
                                 <td id="td-9">2.266</td>
                                 <td id="td-10">0.369
                                    
                                 </td>
                              </tr>
                              <tr id="tr-4">
                                 <td id="td-11">M</td>
                                 <td id="td-12">2</td>
                                 <td id="td-13">−0.041</td>
                                 <td id="td-14">2.380</td>
                                 <td id="td-15">−0.017<sup>ns</sup></td>
                              </tr>
                              <tr id="tr-5">
                                 <td id="td-16">N</td>
                                 <td id="td-17">2</td>
                                 <td id="td-18">0.055</td>
                                 <td id="td-19">0.706</td>
                                 <td id="td-20">0.078
                                    
                                 </td>
                              </tr>
                              <tr id="tr-6">
                                 <td id="td-21">P</td>
                                 <td id="td-22">5</td>
                                 <td id="td-23">0.463</td>
                                 <td id="td-24">2.154</td>
                                 <td id="td-25">0.215
                                    
                                 </td>
                              </tr>
                              <tr id="tr-7">
                                 <td id="td-26">S</td>
                                 <td id="td-27">4</td>
                                 <td id="td-28">0.146</td>
                                 <td id="td-29">1.629</td>
                                 <td id="td-30">0.090</td>
                              </tr>
                           </tbody>
                        </table>
</div>

</section>
<section class='' data-state='' id='slide-27'>
  <h2>Testing for founder effects</h2>
  <ul>
<li>Under neutrality, for a constant population size at equilibrium:</li>
</ul>

<p><img src="assets/img/tajdeqn.png" style="border: 0px;" width="40%"></p>

<ul>
<li>Tajima&#39;s D compares the estimates of $\theta$ obtained from pairwise distances and from segregating sites:

<ul>
<li>$D&lt;0$: e.g. population bottleneck</li>
<li>$D=0$: e.g. constant population size</li>
<li>$D&gt;0$: e.g. population contraction</li>
</ul></li>
</ul>

</section>
<section class='' data-state='' id='slide-28'>
  <h2>Evidence for local founder effects</h2>
  <p><img src="assets/img/tajd.jpg" style="border: 0px;" width="45%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. (2001) PNAS</em></p>
</div>

</section>
<section class='' data-state='' id='slide-29'>
  <h2>Transmission and reversion of drug resistance</h2>
  <p><img src="assets/img/transmittedresistance.jpg" style="border: 0px;" width="80%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Little, Frost et al. J. Virol. (2008)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-30'>
  <h2>Reversion of transmitted resistance</h2>
  <p><img src="assets/img/k103n.jpg" style="border: 0px;" width="80%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Little, Frost et al. J. Virol. (2008)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-31'>
  <h2>Resistant mutants are &#39;fit&#39;</h2>
  <p><img src="assets/img/k103rc.jpg" style="border: 0px;" width="70%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Little, Frost et al. J. Virol. (2008)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-32'>
  <h2>Summary</h2>
  <ul>
<li>Acquired resistance can emerge rapidly

<ul>
<li>Pre-existing, but variable, resistant virus</li>
<li>Decreasing wild-type viruses results in more target cells, facilitating the emergence of resistance</li>
</ul></li>
<li>Transmitted fitness may revert slowly</li>
<li>Given rapid escape and slow reversion, we would expect transmitted resistance to be very common

<ul>
<li>Transmitted resistance is rarer than expected, possibly due to lower transmission rates</li>
</ul></li>
</ul>

</section>
<section class='' data-state='' id='slide-33'>
  <h2>Escape from immune responses</h2>
  <ul>
<li>In addition to selection by antiviral agents, there is also selection from the adaptive immune response

<ul>
<li>Humoral (antibody) responses</li>
<li>Cellular (cytotoxic T lymphocyte, CTL) responses</li>
</ul></li>
<li>What are the dynamics of escape <em>within</em> infected individuals?</li>
<li>How is this affected by transmission <em>between</em> individuals?</li>
</ul>

</section>
<section class='' data-state='' id='slide-34'>
  <h2>Dynamics of antibody responses</h2>
  <p><img src="assets/img/abreview.png" style="border: 0px;" width="60%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. Curr. Opin. HIV AIDS (2008)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-35'>
  <h2>Measuring antibody responses</h2>
  <p><img src="assets/img/monogram.png" style="border: 0px;" width="80%"></p>

</section>
<section class='' data-state='' id='slide-36'>
  <h2>Within-host antibody responses</h2>
  <p><img src="assets/img/neut2.png" style="border: 0px;" width="50%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. PNAS (2005)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-37'>
  <h2>Variation between individuals</h2>
  <p><img src="assets/img/neut3.png" style="border: 0px;" width="85%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Frost et al. PNAS (2005)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-38'>
  <h2>Modelling escape from antibody responses</h2>
  <ul>
<li>It was long perceived that antibodies are ineffective in controlling HIV</li>
<li>Some individuals <em>do</em> mount powerful responses

<ul>
<li>Too little, too late</li>
</ul></li>
</ul>

<p><img src="assets/img/hs97.png" style="border: 0px;" width="50%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Haraguchi and Sasaki (1997)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-39'>
  <h2>Model results</h2>
  <p><img src="assets/img/modelneut.png" style="border: 0px;" width="50%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Little, Frost et al. PNAS (2005)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-40'>
  <h2>Cross-reactivity and escape</h2>
  <p><img src="assets/img/modelneut2.png" style="border: 0px;" width="85%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Little, Frost et al. PNAS (2005)</em></p>
</div>

</section>
<section class='' data-state='' id='slide-41'>
  <h2>Cellular responses</h2>
  <ul>
<li>Another important arm of the adaptive immune response is the cytotoxic T lymphocyte (CTL) response</li>
<li>CTLs recognise infected cells, as these cells present viral peptides at their surface</li>
<li>HIV can escape specific CTL responses through a small number of mutations</li>
<li>On transmission with an escape mutant:

<ul>
<li>Mutant will revert back to wild type (if mutant is not recognised)</li>
<li>Mutant will remain (if mutant is recognised)</li>
</ul></li>
<li>These lead to complex dynamics, depending on the transmission rate</li>
</ul>

</section>
<section class='' data-state='' id='slide-42'>
  <h2>CTL responses and escape</h2>
  <p><img src="assets/img/poon2007fig2.png" style="border: 0px;" width="85%"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Poon et al. PLoS Path 2007</em></p>
</div>

</section>
<section class='' data-state='' id='slide-43'>
  <h2>Model predictions</h2>
  <ul>
<li>Evolution at the population level reflects the &#39;averaging&#39; over multiple individuals</li>
<li>Variation within the individual is affected by the transmission rate</li>
</ul>

</section>
<section class='' data-state='' id='slide-44'>
  <h2>Capturing selection pressure</h2>
  <ul>
<li>A common bioinformatic approach to detecting selection is to compare $dN$ and $dS$:

<ul>
<li>$dN$: the rate of <em>nonsynonymous</em> or amino acid changing mutations</li>
<li>$dS$: the rate of <em>synonymous</em> or amino acid preserving mutations</li>
</ul></li>
<li>The relative rates of $dN$ and $dS$ are informative about different types of selection</li>
</ul>

</section>
<section class='' data-state='' id='slide-45'>
  <h2>Capturing within- and between-host variation</h2>
  <ul>
<li>Between-host variation:

<ul>
<li>$dN$ and $dS$ counted along the viral phylogeny</li>
</ul></li>
<li>Within-host variation

<ul>
<li>Compare nonsynonymous and synonymous <em>mixtures</em></li>
</ul></li>
</ul>

</section>
<section class='' data-state='' id='slide-46'>
  <h2>What are mixtures?</h2>
  <ul>
<li>When we sequence a sample of blood from an infected person, we actually sequence a <em>swarm</em> of different variants</li>
<li>Variation in this swarm can be detected by mixtures</li>
</ul>

<p><br></p>

<p><img src="assets/img/mix.png" width="40%" style="border:0px;"></p>

</section>
<section class='' data-state='' id='slide-47'>
  <h2>Why are mixtures important?</h2>
  <p><img src="assets/img/mixsweep.png" width="70%" style="border: 0px;"></p>

</section>
<section class='' data-state='' id='slide-48'>
  <h2>Selection pressures within and between hosts</h2>
  <p><img src="assets/img/hivrtdnds.png" width="50%" style="border: 0px;"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Poon et al. PLoS Path 2007</em></p>
</div>

</section>
<section class='' data-state='' id='slide-49'>
  <h2>Selection on CTL epitopes</h2>
  <p><img src="assets/img/hivrtmix.png" width="80%" style="border: 0px;"></p>

<div style="font-family: Arial; font-size: 12px;" align="right">
<p><em>Poon et al. PLoS Path 2007</em></p>
</div>

</section>
<section class='' data-state='' id='slide-50'>
  <h2>Conclusions</h2>
  <ul>
<li>Our ability to understand what goes on <em>within</em> the HIV infected individual is attributable to increased data

<ul>
<li>Availability of therapy</li>
<li>Sequence data</li>
<li>Phenotypic data</li>
<li>Modeling has moved from being &#39;data-free&#39; to one which is data-intensive</li>
</ul></li>
<li>HIV evolution and dynamics occurs at the &#39;front line&#39; within the individual, and variation at the population level reflects averaging over the selection pressures from many hosts</li>
</ul>

</section>
<section class='' data-state='blackout' id='slide-51'>
  
  <h2 style="color:#FFFFFF">Acknowledgements</h2>

<div style="color:#FFFFFF">

- Too many to list, but especially
  - Angela McLean
  - Sergei Kosakovsky Pond
  - Art Poon
  - Doug Richman
  
</div>

<p><br></p>

<p><h3 style="color:#FFFFFF">Funding</h2></p>

<p><img src="assets/img/mrc.png" width="20%" style="border: 0px;"></p>

</section>
    </div>
  </div>
</body>
  <script src="libraries/frameworks/revealjs/lib/js/head.min.js"></script>
  <script src="libraries/frameworks/revealjs/js/reveal.min.js"></script>
  <script>
  // Full list of configuration options available here:
  // https://github.com/hakimel/reveal.js#configuration
  Reveal.initialize({
    controls: true,
    progress: true,
    history: true,
    center: false,
    theme: Reveal.getQueryHash().theme || 'simple', 
    transition: Reveal.getQueryHash().transition || 'linear', 
    dependencies: [
    // Cross-browser shim that fully implements classList -
    // https://github.com/eligrey/classList.js/
      { src: 'libraries/frameworks/revealjs/lib/js/classList.js', condition: function() { return !document.body.classList;}},
      // Zoom in and out with Alt+click
      { src: 'libraries/frameworks/revealjs/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
      // Speaker notes
      { src: 'libraries/frameworks/revealjs/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },
      // Remote control your reveal.js presentation using a touch device
      //{ src: 'libraries/frameworks/revealjs/plugin/remotes/remotes.js', async: true, condition: function() { return !!document.body.classList; } }
      ]
  });
  </script>  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      processEscapes: true
    }
  });
</script>
<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script> -->
<script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<!-- LOAD HIGHLIGHTER JS FILES -->
<script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<!-- DONE LOADING HIGHLIGHTER JS FILES -->
 

</html>